### PR TITLE
Support propagation of cluster-scoped resources

### LINKED ISSUE #306: Add Support for Cluster-Scoped Objects
Federation-v2 currently supports namespace-scoped Kubernetes objects (i.e. ConfigMap). Cluster-scoped objects (i.e. ClusterRole) can not be [added](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/development.md#adding-a-new-api-type) due to how fed-v2 version propagation currently works. The UX for adding cluster-scoped objects was addressed by https://github.com/kubernetes-sigs/federation-v2/issues/242.

### PR DESCRIPTION
**Pull request by marun**
Adds a cluster-scoped version adapter and validates propagation in e2e with a generated cluster-scoped CRD.

Fixes #306

### CODE REVIEW

**Comment by marun** [comment]
Yuck, github doesn't order the commits at all.  I think I might break this apart to make it easier to review.

**Comment by marun** [comment]
Please review #334 and #335 and when they've merged and this PR is rebased, this PR should be much easier to reason about.

**Comment by danehans** [comment]
This has a merge conflict with #335 b/c both add `test/e2e/crd.go`.

**Comment by marun** [comment]
@danehans That's be design, since it includes #335 in its entirety.  As per the first comment, this PR will be rebased against master once the PRs it is based on have merged.

File: test/e2e/version.go [hunk 1]
```diff
@@ -310,3 +459,16 @@ func removeOneCluster(clusterNames []string, versionMap map[string]string) ([]st
 	delete(versionMap, targetName)
 	return clusterNames, versionMap
 }
+
+func inSupportedScopeIt(description string, namespaced bool, f interface{}) {
+	// TODO(marun) Refactor test to support skipping before setup so
+	// that skip details can be logged.  The implicit skipping this
+	// function performs doesn't provide a good indication of which
+	// tests are skipped and why.
+	if namespaced && framework.TestContext.LimitedScope {
```
[the remark this answers is not in the document]
Comment (marun) [reply test/e2e/version.go]: Oof, good catch!

File: test/common/testobjects.go [hunk 2]
```diff
@@ -111,8 +115,10 @@ func GetPlacementTestObject(typeConfig typeconfig.Interface, namespace string, c
 	if err != nil {
 		return nil, err
 	}
+	if typeConfig.GetNamespaced() || typeConfig.GetTemplate().Kind == util.NamespaceKind {
```
[the remark this answers is not in the document]
Comment (marun) [reply test/common/testobjects.go]: Done

**Comment by danehans** [comment]
I tested this PR and it works as described. /lgtm

**Pull request merged**